### DISCUSSION TITLE
Seeking Feedback: Remote Error Logging

### DISCUSSION THREAD

**Discussion by chihsuan** [woocommerce/woocommerce#51568] [Request for Comments]
We know many of you have faced frustrating errors and stability issues with WooCommerce. To address these ongoing challenges, we have been working on many improvements to WooCommerce's stability and reliability. One of these improvements is the new Remote Logging feature, which will be introduced in WooCommerce 9.4. 

## The Problem We're Solving

- Frequent unexpected errors disrupting store operations
- Difficulty in reproducing and fixing issues across different environments
- Delays in identifying and resolving critical bugs

## How Remote Logging Helps

- Automatic Error Capture: Logs PHP and JavaScript errors in real-time
- Comprehensive Error Analysis: Allows our team to spot patterns and prioritize fixes
- Proactive Issue Resolution: Helps us address problems before they impact more users

## Important Notes on Privacy and Consent

- The remote logs are only accessible to WooCommerce and Automattic teams.
- Remote Logging is strictly opt-in. It only activates when users have explicitly agreed to tracking. For detailed information about what data is collected and how it's used, please visit our [Usage Tracking info page](https://woocommerce.com/usage-tracking/).
- If you want to allow other Tracking features but only disable remote error logging, you can set the `woocommerce_feature_remote_logging_enabled` option to `no` to disable this feature specifically.

## For Developers

Developers interested in the technical implementation can refer to the [Remote Error Logging Epic](https://github.com/woocommerce/woocommerce/issues/49276) to see how we implemented this feature.

## Your Feedback

We'd love to hear your thoughts on this new feature. If you have any questions or concerns, please share them in the comments below. Thanks for helping us make WooCommerce better!

**Comment by mcarter450** [comment]
I noticed this feature was turned on in both of my Woo production sites, and I had never enabled it, not to my knowledge. I don't like the idea of remote tracking of PHP and JavaScript errors for my self-hosted app.

**Reply by chihsuan** [reply]
@mcarter450  Hi there! Thank you for reaching out and sharing your feedback. I understand you don't like the idea of remote tracking of PHP and JavaScript errors for your self-hosted app.

The remote logging feature only sends logs if you opt-in to usage tracking during the onboarding process or via `WooCommerce > Settings > Advanced > WooCommerce.com`. If you don't opt-in, no data will be sent (even if the feature is enabled). We understand that the UI might not be very clear about this, and we've made improvements in this regard. You can expect these changes to be available in version 9.8, as detailed in https://github.com/woocommerce/woocommerce/pull/55119.


You can also check out this article for more details: https://developer.woocommerce.com/2024/09/23/recent-updates-to-error-handling-and-optional-remote-error-logging/
